### PR TITLE
collisions

### LINKED ISSUE #31: collision with map
once mapgen

### PR DESCRIPTION
**Pull request by jesings**

### CODE REVIEW

File: src/map/grid.rs [hunk 1]
```diff
@@ -15,3 +15,18 @@ pub struct Grid {
     pub rows: usize,
     pub tiles: Vec<Arc<Tile>>,
 }
+
+impl Grid {
+    pub fn passable(&self, tilepos: (isize, isize)) -> bool {
+        if tilepos.0 < 0 || tilepos.1 < 0 {
```
Comment (anotherLostKitten) [inline src/map/grid.rs:21]: hmm i know is this was ASSembly we could just to an unsigned compare to get whether we're in the map bounds is there a way to do this in ruust

File: src/class/void_orifice.rs [hunk 2]
```diff
@@ -31,9 +28,12 @@ impl VoidOrifice {
         Some(move |gamedata: Arc<Gamedata>, sender: &mpsc::Sender<PktPayload>| {
             if offcd {
                 let pl = &mut gamedata.players[pid].lock().unwrap();
-                // v-w-y <- wubbles the funny collision check
-                pl.pos = (pl.pos.0 + dir.0, pl.pos.1 + dir.1);
-                sender.send(PktPayload::Delta(vec!(DeltaEvent{pid: pid, poschange: dir}))).unwrap();
+                let pnp = (pl.pos.0 + dir.0, pl.pos.1 + dir.1);
+                if gamedata.grid.passable(pnp) {
+                    // v-w-y <- wubbles the funny collision check :D
```
Comment (anotherLostKitten) [inline src/class/void_orifice.rs]: considering v-w-y now exists i think u can delet the placceholder comment

File: src/map/grid.rs (hunk 1 above)
Comment (anotherLostKitten) [reply src/map/grid.rs:21]: ~~that said considering that with proper map gen we shouldn't even need to concern ourselves with going out of bounds, as there should always be some wall bordering which we can actually draw, also how necessary is this comparison to begin with?~~

Comment (anotherLostKitten) [reply src/map/grid.rs:21]: i dont think we will ever want the map to directly touch the screen edge

Comment (anotherLostKitten) [reply src/map/grid.rs:21]: well actually we will prolly add destructible enviornments so nvm disregard that point

Comment (jesings) [reply src/map/grid.rs:21]: Ok disregarded.

Comment (anotherLostKitten) [reply src/map/grid.rs:21]: no i just meant the point about whether they were necessary my question about doing an unsigned compare stands

File: src/map/grid.rs [hunk 3]
```diff
@@ -15,3 +15,18 @@ pub struct Grid {
...
+            return false;
+        }
+        let p0 = tilepos.0 as usize;
+        let p1 = tilepos.1 as usize;
```
Comment (anotherLostKitten) [inline src/map/grid.rs:25]: we are already converting to usize, so we could do an unsigned comppare against the collum length, in which case comparing to zero would not be necessary

Comment (anotherLostKitten) [reply src/map/grid.rs:25]: @jesings ^

File: src/map/grid.rs [hunk 4]
```diff
@@ -15,3 +15,18 @@ pub struct Grid {
...
+        if p0 >= self.cols || p1 >= self.rows {
```
Comment (anotherLostKitten) [inline src/map/grid.rs:26]: if it is negative, the unsigned value will already b larger
do u get it now

**Pull request merged**